src/Sulu/Bundle/AdminBundle/Resources/public/js/components/login/main.js

```diff
@@ -1,54 +1,709 @@
+                             '  <span class="'+ constants.websiteSwitchClass +'"><%= backWebsiteMsg %></span>',
+                             '  <span class="'+ constants.loginRouteClass +'"><%= loginRouteMsg %></span>',
+                             '</div>'].join('')
         },
 
         /**
          * trigger after initialization has finished
          *
-         * @event sulu.login.initialized
+         * @event sulu.login.[INSTANCE_NAME].initialized
          */
-            INITIALIZED = function() {
+        INITIALIZED = function() {
             return createEventName.call(this, 'initialized');
-        };
+        },
 
+        createEventName = function(postfix) {
+            return 'sulu.login.' + ((!!this.options.instanceName) ? this.options.instanceName + '.' : '') + postfix;
+        };
 
     return {
 
         /**
-         * Initialize the component
+         * Initializes the component
          */
         initialize: function() {
-            // todo: login
+            // merge defaults
+            this.options = this.sandbox.util.extend(true, {}, defaults, this.options);
+            this.initProperties();
+            this.render();
+            if (this.options.resetMode === false) {
+                this.bindDomEvents();
+            } else {
+                this.bindResetDomEvents();
+            }
+            this.focusUsername();
             this.sandbox.emit(INITIALIZED.call(this));
+        },
+
+        /**
+         * Initializes the component properties
+         */
+        initProperties: function() {
+            this.dom = {
+                $bg: null,
+                $box: null,
+                $frames: null,
+                $login: null,
+                $resetMail: null,
+                $forgotSwitch: null,
+                $loginSwitch: null,
+                $loginBtn: null,
+                $resetMailBtn: null,
+                $resendBtn: null,
+                $loginForm: null,
+                $loader: null,
+                $reset: null
+            };
+            this.mouseOrigin = {
+                x: 0,
+                y: 0
+            };
+            this.resetMailUser = null;
+            this.setMovementRatio();
+        },
+
+        /**
+         * Calcualtes ratios which tells how much pixel the background moves per pixel the mosue moved
+         */
+        setMovementRatio: function() {
+            this.movementRatio = {
+                x: this.options.shiftSpace/this.sandbox.dom.width(this.sandbox.dom.$window),
+                y: this.options.shiftSpace/this.sandbox.dom.height(this.sandbox.dom.$window)
+            };
+        },
+
+        /**
+         * Renders the component
+         */
+        render: function() {
+            this.sandbox.dom.addClass(this.$el, constants.componentClass);
+            this.renderBg();
+            this.renderBox();
+        },
+
+        /**
+         * Render background
+         */
+        renderBg: function() {
```

Comment:
The same is valid for all other functions and variables containing abbreviations.